### DISCUSSION TITLE
faiss Kmeans .obj attribute

### DISCUSSION THREAD

**Discussion by TopoKunst** [facebookresearch/faiss#3561] [Q&A]
# Summary

Hello,

I have tried to use faiss on clustering problems with faiss.Clustering, and I want to measure the result by some metric. I notice that faiss.Kmeans has an attribute .obj in a [notebook file](https://gist.github.com/mdouze/9eb96d941c94ef59482a069e5862a650) of this repo, but I fail to get it with faiss.Clustering.

I am wondering if faiss.Clustering has this property or some other metrics? Or, could you please tell me how kmeans.obj is computed?

Thank you

# Platform

OS: 

Faiss version: 

Installed from:  

Faiss compilation options: 

Running on:
- [x] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [x] Python

**Comment by mdouze** [comment]
please use Clustering::iteration_stats

**Comment by TopoKunst** [comment]
I use `clus.iteration_stats` in Python where clus is a `faiss.Clustering` object, but it reports AttributeError: 'Clustering' object has no attribute 'iteration_stats'. 

I read the changelog and find the iteration_stats is added in 1.6.3, while my Faiss version is 1.5.1. Is this the reason that I cannot use it?

My environment does not support conda installation, so I can only use faiss 1.5.1 instead of faiss-cpu or faiss-gpu. Could you pls tell me how this metric is computed? I read faiss/Clustering.cpp and guess it is kind of like accumulative square error between each sample and its centroid?
> please use Clustering::iteration_stats

**Comment by TopoKunst** [comment]
Sorry that I misclose the issue, pls check my first reply.
